### PR TITLE
Rating component

### PR DESCRIPTION
**Pull request by rpinoon**
Disclaimer: pls disregard the 'react-star-rating-component'. Ended up not using it,

I'm making a star-rating component here that sends an update request to the backend that way when the Finished List fetches the books, their individual rating goes with them and is displayed properly.

### CODE REVIEW

File: components/bookshelf/Book.tsx [hunk 1]
```diff
@@ -56,6 +69,16 @@ const Book = ({ book, reading = false }: BookProps) => {
         <BookContent>
           <HeaderBar>
             <Title>{book.title}</Title>
+            {(status === StatusEnum.reading || section === "finished") && (
+              <>
+                <Rating
+                  name="simple-controlled"
+                  size="small"
+                  value={currentRating}
+                  onChange={rateBook}
```
Comment (normancapule) [inline components/bookshelf/Book.tsx:82]: Don't forget to disable this button when the API request is loading. Else your client may create x number of requests to the backend by clicking the button x number of times. 

Take note of this - one of the things that QAs will test your future features.

**Review by normancapule — APPROVED** [review]

File: api/books/setRating.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+          rating: rating,
+        },
+      },
+    })
+    .then((res) => {
+      const book = res.data
+      return book
+    })
+    .catch((error) => {
+      console.log(error)
```
Comment (mikodagatan) [inline api/books/setRating.ts]: Can you apply the error handling code you just made here? Thanks man

File: components/bookshelf/Book.tsx [hunk 3]
```diff
@@ -42,6 +46,15 @@ const Book = ({ book, reading = false }: BookProps) => {
     removeFromList(book.user_book_id)
   }
 
+  const unmarkBook = () => {
+    console.log("book unmarked")
```
Comment (mikodagatan) [inline components/bookshelf/Book.tsx:51]: Can you put an action instead of console log here?

File: components/bookshelf/Book.tsx (hunk 1 above)
Comment (rpinoon) [reply components/bookshelf/Book.tsx:82]: copy that. will take note. thanks!

File: api/books/setRating.ts (hunk 2 above)
Comment (rpinoon) [reply api/books/setRating.ts]: sure thing. its just that the error handling is still unmerged in a different branch thats why I wasn't able to use it here 😅

File: components/bookshelf/Book.tsx (hunk 3 above)
Comment (rpinoon) [reply components/bookshelf/Book.tsx:51]: yep. will do on a seperate PR

**Pull request merged**